src/mbgl/renderer/tile_mask.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <mbgl/tile/tile_id.hpp>
+
+#include <set>
+
+namespace mbgl {
+
+// A TileMask is a set of TileIDs that describe what part of a tile should be rendered. It omits
+// those parts of the tile that are covered by other/better tiles. If the entire tile should be
+// rendered, it contains the { 0, 0, 0 } tile. If it's empty, no part of the tile will be rendered.
+// TileMasks are typically generated with algorithm::updateTileMasks().
```

Comment:
The key to understanding how the TileMask is generated and used is to understand the relationship between:

* The tile ID of the tile being rendered
* The tile IDs of the renderable descendants of that tile
* The tile IDs of elements of the resulting TileMask

Can you spell out that relationship here or elsewhere, with either an algebraic definition or example?